lib/ansible/module_utils/common/collections.py

```diff
@@ -0,0 +1,31 @@
+# Copyright (c), Sviatoslav Sydorenko <ssydoren@redhat.com> 2018
+# Simplified BSD License (see licenses/simplified_bsd.txt or https://opensource.org/licenses/BSD-2-Clause)
+"""Collection of low-level utility functions."""
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+# Note: When getting Sequence from collections, it matches with strings.  If
+# this matters, make sure to check for strings before checking for sequencetype
+try:
+    """Python 3.3+ branch."""
+    from collections.abc import KeysView, Sequence
+except ImportError:
+    """Use old lib location under 2.6-3.2."""
+    from collections import KeysView, Sequence
+
+SEQUENCETYPE = frozenset, KeysView, Sequence
+
+
+from ansible.module_utils.six import binary_type, text_type
+
+
+def is_sequence(seq, include_strings=False):
+    """Identify whether the input is a sequence.
+
+    Strings and bytes are not sequences here.
+    """
+    is_string_type = isinstance(seq, (text_type, binary_type))
+    maybe_a_sequence = not is_string_type or include_strings
+    return maybe_a_sequence and isinstance(seq, Sequence)
```

Comment:
Style-wise, we would prefer that this uses an if statement rather than ands and ors. if statements make it very clear what the purpose of the code is whereas raw ands and ors being saved to variables muddy the waters.

Reply:
Well, that's why I added a named variables for conditions. This also make it easily readable.